### PR TITLE
 Preserve Original Tensor Model Output in RepaintPipeline

### LINKED ISSUE #7545: RepaintPipeline: Preserve Original Tensor Model Output
I am trying to use the output of the RepaintPipeline in a torch.tensor, but the inherent scaling and conversion to PIL Image or numpy makes assumptions about the original normalization used for the input model.

Add extra case for non-rescaled tensor passed to ImagePipelineOutput in repaint_pipeline.py.

I have casted between PIL, np.array, and tensor, but the lack of preservation of -1 to 0 leads to inconsistencies in rescaling.

### PR DESCRIPTION
**Pull request by c-merchant**
This PR adds an extra case for the original, non-rescaled tensor output passed to ImagePipelineOutput in repaint_pipeline.py. By not rescaling between 0 and 1 (as opposed to say -1 and 1), it doesn't make assumptions about the original normalization used for the model. The changes preserve the logic of the original implementation but add a case to prevent this rescaling. The tensor output is still cast to numpy for compatibility with ImageOutputPipeline. I have also added this change to the documentation.

Fixes #7545

## Before submitting
This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
 Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
 Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
 Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sayakpaul @yiyixuxu @DN6

### CODE REVIEW

**Review by yiyixuxu — COMMENTED** [review]
thanks for the PR, however, this is inconsistent with the rest of our pipelines 

the output_types we support are: `"latent", "pt", "np", "pil"` -  `pt` is same as `Torch.tensor` but it should be normalized the same as np output; feel free to refactor it with the VaeImageProcessor (https://github.com/huggingface/diffusers/blob/7956c36aaa2b43584f16f7e755e46aa2d7efa721/src/diffusers/image_processor.py#L734) similar to what's done in other pipelines, e.g https://github.com/huggingface/diffusers/blob/7956c36aaa2b43584f16f7e755e46aa2d7efa721/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L1026

**Comment by yiyixuxu** [comment]
close this now since:
1. it is a deprecated pipeline
2.the change is not consistent with other pipelines
